zulipterminal/ui_tools/buttons.py

```diff
@@ -268,6 +269,35 @@ def mark_muted(self) -> None:
     # TODO: Handle event-based approach for topic-muting.
 
 
+class EmojiButton(TopButton):
+    def __init__(self, controller: Any, width: int, emoji_name: str,
+                 message: Message) -> None:
+        self.controller = controller
+        self.emoji_name = emoji_name
+        self.message = message
+        super().__init__(
+            controller=controller,
+            caption=emoji_name,
+            prefix_character='',
+            show_function=controller.model.toggle_message_reaction,
+            width=width)
+        if self.user_has_reacted_to_msg():
+            self.update_widget(('bar', ' ✓ '), 'bar')
+
+    def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
+        if is_command_key('ENTER', key):
+            # Note that this is called before toggle_message_reaction.
+            if self.user_has_reacted_to_msg():
```

Comment:
[minor]: This could be shortened with comprehension :sweat_smile:

Reply:
Are you referring to list comprehension, because I don't see any lists here? :thinking:
Or, you are speaking about the function user_has_reacted_to_msg?

Reply:
I was actually referring to the if/else logic below. Sorry for confusion :)

My point was that we could shorten the condition by doing something like:

```python
self.update_widget(('bar', '' if self.user_has_reacted_to_msg else CHECK_MARK), 'bar')
```

Reply:
We could surely do that. But, I think this would be a bit difficult to read and interpret. Maybe, we should wait for Neil's review on this?